### PR TITLE
Add initial support for managed instances

### PR DESCRIPTION
**Pull request by tenyo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,14 +8,22 @@ This API provides simple restful API access to SpotInst Spot market services.
...
+
 GET /v1/spot/{account}/elastigroups
 POST /v1/spot/{account}/elastigroups
 PUT /v1/spot/{account}/elastigroups/{elastigroup}
 DELETE /v1/spot/{account}/elastigroups/{elastigroup}
+
+GET /v1/spot/{account}/managedinstances
+POST /v1/spot/{account}/managedinstances
+PUT /v1/spot/{account}/managedinstances/{managedinstance}
+DELETE /v1/spot/{account}/managedinstances/{managedinstance}
```
Comment (fishnix) [inline README.md]: Let's just call this `instances`

File: api/handlers_managedinstances.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/YaleSpinup/spot-api/apierror"
```
Comment (fishnix) [inline api/handlers_managedinstances.go]: migrate over to the shared `apierror` package.  it should be a drop-in replacement.

**Comment by fishnix** [comment]
This looks good - are we going to need a keypair?

Also we should consider if we want to map the types at the `api` package and how my different instance sizes we want.... the API should probably determine that list.

**Comment by fishnix** [comment]
👍

**Comment by tenyo** [comment]
What do you mean by keypair? We just need the account id and token for each Spot account

**Comment by fishnix** [comment]
I see the `keypair` in the input - what about userdata too?

**Comment by tenyo** [comment]
These are the same ones we use for EC2 instances, so no keypair and whatever userdata we usually have for Linux and Windows

**Pull request merged**